### PR TITLE
add simple perf test for getdeviceid

### PR DESCRIPTION
**Pull request by MarkZuber**

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]

File: msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        public void ValidateGetDeviceIdPerformance()
+        {
+            var platformProxy = PlatformProxyFactory.GetPlatformProxy();
+            // Call it once to pre-load it.  We're not worried about the time it takes to call it
+            // the first time, we're worried about subsequent calls.
+            platformProxy.GetDeviceId();
+
+            using (new PerformanceValidator(5, "GetDeviceId"))
+            {
+                platformProxy.GetDeviceId();
```
Comment (henrik-me) [inline msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs]: >platformProxy.GetDeviceId(); [](start = 16, length = 28)

consider calling all the properties on the platformproxy and broaden the scope of the test a bit

File: msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using Microsoft.Identity.Core;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Test.MSAL.NET.Common.Core.Helpers;
+
+namespace Test.MSAL.NET.Unit.net45
```
Comment (henrik-me) [inline msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs]: >Test.MSAL.NET.Unit.net45 [](start = 10, length = 24)

should imo have the same test in core... we know right now that it is performing better, however perhaps we start introducing something there.

File: msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+{
+    [TestClass]
+    public class PlatformProxyTests
```
Comment (henrik-me) [inline msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs]: >PlatformProxyTests [](start = 17, length = 18)

consider: PlatformProxyPerformanceTests

File: msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        [TestMethod]
+        public void ValidateGetDeviceIdPerformance()
+        {
+            var platformProxy = PlatformProxyFactory.GetPlatformProxy();
```
Comment (henrik-me) [inline msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs]: >var platformProxy = PlatformProxyFactory.GetPlatformProxy(); [](start = 12, length = 60)

Consider even measuring this

File: msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs (hunk 2 above)
Comment (bgavrilMS) [reply msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs]: All unit tests defined in the net45 assemblies are referenced in the netcore assembly and run there as well, unless you add a precompilation #if !NET_CORE 

But I agree that the default namespace is a bit confusing, should be changed to Test.MSAL.NET.Unit

**Review by henrik-me — CHANGES_REQUESTED** [review]
:clock1:

File: msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs (hunk 3 above)
Comment (MarkZuber) [reply msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs]: done.

File: msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs (hunk 1 above)
Comment (MarkZuber) [reply msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs]: fixed

File: msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs (hunk 4 above)
Comment (MarkZuber) [reply msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs]: fixed

**Review by henrik-me — APPROVED** [review]
:shipit:

**Pull request merged**